webservices/resources/legal.py

```diff
@@ -30,21 +71,32 @@ def get(self, q, from_hit=0, hits_returned=20, _type='all', **kwargs):
         else:
             types = [_type]
 
+        parsed_query = parse_query_string(q)
+        terms = parsed_query.get('terms')
+        phrases = parsed_query.get('phrases')
+
         results = {}
         total_count = 0
         for _type in types:
             query = {"query": {"bool": {
-                     "must": [{"match": {"_all": q}}, {"term": {"_type": _type}}],
-                               "should": [{"match": {"no": q}},
-                                               {"match_phrase": {"_all": {"query": q,
-                                                                          "slop": 50}
-                                                                 }
-                                                }]
+                     "must": [
+                         {"term": {"_type": _type}},
+                         ],
+                     "should": [
+                         {"match": {"no": q}},
+                         {"match_phrase": {"_all": {"query": q, "slop": 50}}},
+                         ]
                      }},
                 "highlight": {"fields": [{"text": {}},
                     {"name": {}}, {"number": {}}]},
                 "_source": {"exclude": "text"}}
 
+            if len(terms):
+                query['query']['bool']['must'].append({"match": {"_all": ' '.join(terms)}})
+
+            for phrase in phrases:
+                query['query']['bool']['must'].append({"match_phrase": {"text": phrase}})
```

Comment:
@adborden Nice work! After playing around with it, this looks right to me. I thought it would capture things like "electronic form filing" as well (i.e., terms in right order but which have other words in between), but it doesn't. So 🚀 .